### PR TITLE
bump to 5.2 and fix a check that always returns null

### LINKED ISSUE #1: Bukkit Server#getOfflinePlayer(String) always returns non-null
https://github.com/LuckPerms/api-cookbook/blob/master/src/main/java/me/lucko/lpcookbook/commands/AddPermissionCommand.java#L39

```
        // Get an OfflinePlayer object for the player
        OfflinePlayer player = this.plugin.getServer().getOfflinePlayer(playerName);

        // Player not known?
        if (player == null) {
            sender.sendMessage(ChatColor.RED + playerName +  " has never joined the server!");
            return true;
        }
```

I think this null check would never evaluate true because getOfflinePlayer(String) always returns non-null. Maybe use OfflinePlayer#hasPlayedBefore() instead of checking for null?

### PR DESCRIPTION
**Pull request by powercasgamer**
Closes #1

### CODE REVIEW

**Comment by lucko** [comment]
Despite what the annotations on newer versions of Bukkit say, I am fairly sure there is still a possibility that the method could return null.

Perhaps `if (player == null || !player.hasPlayedBefore()) {...}`would be better?

**Comment by lucko** [comment]
(happy to be proven wrong on that of course! I might've remembered incorrectly)

**Comment by powercasgamer** [comment]
> Despite what the annotations on newer versions of Bukkit say, I am fairly sure there is still a possibility that the method could return null.
> 
> Perhaps `if (player == null || !player.hasPlayedBefore()) {...}`would be better?

I've done that. Anything else that should be changed?

File: src/main/java/me/lucko/lpcookbook/listener/PlayerFirstJoinListener.java [hunk 1]
```diff
@@ -32,7 +32,7 @@ private void onDataSave(PlayerDataSaveEvent e) {
         Set<Outcome> outcomes = result.getOutcomes();
 
         if (outcomes.contains(Outcome.CLEAN_INSERT)) {
-            Bukkit.broadcastMessage(ChatColor.GREEN + e.getUsername() + " joined the network for the first time!");
+            Bukkit.broadcastMessage(ChatColor.GREEN + e.getUsername() + " joined the server for the first time!"); // made this server instead of network since all the other messages also says server
```
Comment (lucko) [inline src/main/java/me/lucko/lpcookbook/listener/PlayerFirstJoinListener.java]: Could you revert this change?

`network` is significant here -- this event will not fire when the player joins the _server_ for the first time. It is called when they first join any server in the network.

**Review by lucko — CHANGES_REQUESTED** [review]

File: src/main/java/me/lucko/lpcookbook/listener/PlayerFirstJoinListener.java (hunk 1 above)
Comment (powercasgamer) [reply src/main/java/me/lucko/lpcookbook/listener/PlayerFirstJoinListener.java]: fixed

**Review by lucko — APPROVED** [review]

**Comment by lucko** [comment]
Thanks!

**Pull request merged**